src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.alive = [];
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    const isHerbivore = beast instanceof Herbivore;
+
+    if (!isHerbivore || beast.hidden) {
+      return;
+    }
+
+    beast.health -= 50;
+
+    if (beast.health === 0) {
```

Comment:
Use `beast.health <= 0` instead of `beast.health === 0` to account for cases when the health goes below 0.

Reply:
Agree